### ISSUE TITLE
Maybe wrong payout destination in staking.Reward event

### ISSUE THREAD

**Issue by TheGoldenEye** [polkadot-js/api#2592]
It seams, that the 'staking.Reward' contains a wrong reward destination account.
Tested with @polkadot/api v1.32.1

- chain westend
- controller account: 5H6b7Xsinpxag8DXE6gPv1UEaTrmWNkaKwsiTg7kad7JBTSe
- stash account: 5D9x7E3f7VT2RqdpA54V8MJSh5xBr7pUV6bCDmGbpjZ9mi75
- reward destination was set to controller in 1291565-2 https://westend.subscan.io/extrinsic/1291565-2

The 'staking.Reward' event, triggered by the 'staking.payoutStakers' extrinsic, contains the wrong destination account number.  
event.data[0] is the stash account, but should be the controller account
example: block 1961768 eventId 14: https://westend.subscan.io/extrinsic/0xa70cce63da0c111ae529099f0ed7c5ac45aeeee13720acbb18bbe99596c2b139?event=1961768-14

**Comment by jacogr** [comment]
From the Rust code -

```
/// The staker has been rewarded by this amount. [stash, amount]
		Reward(AccountId, Balance),
```

It is always the stash since that is the id for the validator. If you want to know who, looking the reward destination for that block.

TL;DR Operating exactly as intended and specified.

**Comment by TheGoldenEye** [comment]
> If you want to know who, looking the reward destination for that block.
How to get this?

**Comment by jacogr** [comment]
https://polkadot.js.org/api/substrate/storage.html#payee-accountid-rewarddestination

**Comment by TheGoldenEye** [comment]
Hm, and how to get this for the right (historic) block?

**Comment by jacogr** [comment]
The bloch hash where the tx was included.

**Comment by TheGoldenEye** [comment]
Don't understand. The payee(AccountId) method is only valid in the context of a block. Seems this is (by default) the last block of the chain.
How can I get the payout destination for block x (or a given block hash) in the past?

**Comment by jacogr** [comment]
`.at` queries is available on all stoage entries - https://polkadot.js.org/api/start/api.query.other.html#state-at-a-specific-block

**Comment by TheGoldenEye** [comment]
thx, was missing this!!!
